crates/libcontainer/src/process/container_intermediate_process.rs

```diff
@@ -130,10 +132,12 @@ pub fn container_intermediate_process(
                     }
                     if let ContainerType::TenantContainer { exec_notify_fd } = args.container_type {
                         let buf = format!("{e}");
-                        if let Err(err) = write(exec_notify_fd, buf.as_bytes()) {
+                        let exec_notify_fd =
+                            unsafe { std::os::fd::OwnedFd::from_raw_fd(exec_notify_fd) };
+                        if let Err(err) = write(&exec_notify_fd, buf.as_bytes()) {
                             tracing::error!(?err, "failed to write to exec notify fd");
                         }
-                        if let Err(err) = close(exec_notify_fd) {
+                        if let Err(err) = close(exec_notify_fd.as_raw_fd()) {
                             tracing::error!(?err, "failed to close exec notify fd");
                         }
                     }
```

Comment:
Previously, as everything was `rawFd`, we would manually close it in the last close call. However, now we are converting it to the `ownedFd`, using it for `write` and then doing another manual `close` call on it. As per docs, `OwnedFd` will attempt to close the underlying file description on `Drop`, so here wouldn't that cause to get called `close` twice on same fd? I think that is not error-ing, as the impl of drop [as seen here](https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#174-186) says (essentially) `we ignore errors so to not accidentally close another valid file desc` . Do we calling close explicitly here can run into the same problem?

Should we instead convert to `ownedFd` for the write call, let its drop take care of closing the fd, and remove the close call altogether?

Reply:
Ah yes good point, I missed this. Will look into this. I think letting the drop automatically close the fd would be good, since OwnedFd has that feature built in.